src/main/java/com/_up/megastore/services/implementations/AuthService.java

```diff
@@ -0,0 +1,40 @@
+package com._up.megastore.services.implementations;
+
+import com._up.megastore.controllers.requests.SignUpRequest;
+import com._up.megastore.data.model.User;
+import com._up.megastore.services.interfaces.IAuthService;
+import com._up.megastore.services.interfaces.IUserService;
+import com._up.megastore.services.mappers.UserMapper;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AuthService implements IAuthService {
+
+  private final IUserService userService;
+  private final PasswordEncoder passwordEncoder;
+
+  public AuthService(IUserService userService, PasswordEncoder passwordEncoder) {
+    this.userService = userService;
+    this.passwordEncoder = passwordEncoder;
+  }
+
+  @Override
+  public void signUp(SignUpRequest signUpRequest) {
+    validateSignUpRequest(signUpRequest);
+    User user = createUser(signUpRequest);
+    userService.saveUser(user);
+  }
```

Comment:
Siguiendo un poco lo que decía Eze, esto en realidad debería ser como una especie de pasamano, porque la responsabilidad de creación de un usuario la tiene el UserService

```suggestion
  public void signUp(SignUpRequest signUpRequest) {
    userService.signUp(signUpRequest);
  }
```

¿Por qué esto así? Porque en realidad el día de mañana cuando tengamos el login, en este service vamos a manejar la lógica para generar todo lo que tenga que ver con la autorización de un usuario